wlauto/devices/android/gem5/__init__.py

```diff
@@ -0,0 +1,820 @@
+                  "create a directory in this folder, and will remove it again "
+                  "once the simulation completes."),
+        Parameter('checkpoint_post_boot', kind=bool, default=False,
+                  mandatory=False, override=True, description="This parameter "
+                  "tells Workload Automation to create a checkpoint of the "
+                  "simulated system once the guest system has finished booting."
+                  " This checkpoint can then be used at a later stage by other "
+                  "WA runs to avoid booting the guest system a second time. Set"
+                  " to True to take a checkpoint of the simulated system post "
+                  "boot."),
+        Parameter('run_delay', kind=int, default=0, mandatory=False,
+                  override=True, description="This sets the time that the "
+                  "system should sleep in the simulated system prior to "
+                  "running and workloads or taking checkpoints. This allows "
+                  "the system to quieten down prior to running the workloads. "
+                  "When this is combined with the checkpoint_post_boot"
+                  " option, it allows the checkpoint to be created post-sleep,"
+                  " and therefore the set of workloads resuming from this "
+                  "checkpoint will not be required to sleep.")
+    ]
+
+    # Overwritten from Device. For documentation, see corresponding method in
+    # Device.
+
+    @property
+    def is_rooted(self):
+        # gem5 is always rooted
+        return True
+
+    def __init__(self, **kwargs):
+        self.logger = logging.getLogger('gem5Device')
+
+        super(Gem5Device, self).__init__(**kwargs)
+        self.adb_name = kwargs.get('adb_name') or self.default_adb_name
+        self.binaries_directory = kwargs.get('binaries_directory', self.default_binaries_directory)
+        self.package_data_directory = kwargs.get('package_data_directory', self.default_package_data_directory)
+        self.external_storage_directory = kwargs.get('external_storage_directory',
+                                                     self.default_external_storage_directory)
+        self.logcat_poll_period = kwargs.get('logcat_poll_period')
+
+        # The gem5 subprocess
+        self.gem5 = None
+
+        self.gem5_port = -1
+        self.busybox = None
+        self._is_initialized = False
+        self._is_ready = False
+        self._just_rebooted = False
+        self._is_rooted = None
+        self._available_frequencies = {}
+        self._available_governors = {}
+        self._available_governor_tunables = {}
+        self._number_of_cores = None
+        self._logcat_poller = None
+        self.gem5outdir = os.path.join(settings.output_directory, "gem5")
+        self.stdout_file = None
+        self.stderr_file = None
+        self.stderr_filename = None
+        self.checkpoint = False
+        self.sckt = None
+
```

Comment:
As above, the `kwargs` procesing is not necessary as it will be handled by super's `__init__`.